wren-ui/src/apollo/server/repositories/threadResponseRepository.ts

```diff
@@ -125,6 +126,18 @@ export class ThreadResponseRepository
       .where({ id })
       .update(transformedData)
       .returning('*');
-    return this.transformFromDBData(result);
+    const transformed = this.transformFromDBData(result);
+    return this.transformToThreadResponse(transformed);
+  }
+
+  private transformToThreadResponse(result: any): ThreadResponse {
```

Comment:
I think we can override the `transformFromDBData` method so that the method in the BaseRepository can benefit from this.
```ts
protected override transformFromDBData = (data: any): ThreadResponse => {
    if (!isPlainObject(data)) {
      throw new Error('Unexpected dbdata');
    }
    const camelCaseData = mapKeys(data, (_value, key) => camelCase(key));
    const formattedData = mapValues(snakeCaseData, (value, key) => {
      if (['error', 'detail'].includes(key)) {
        return JSON.parse(value);
      }
      return value;
    });
    return formattedData;
  };
```

Reply:
addressed in 59928af